### DISCUSSION TITLE
Build err

### DISCUSSION THREAD

**Discussion by dabao1955** [Moe-hacker/ruri#5] [Q&A]
```
ruri.c:1021:41: error: use of undeclared identifier '__SNR_futex_waitv'
  seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(futex_waitv), 0);
                                        ^
```

clang version:14.0.1

**Comment by Moe-hacker** [comment]
> ```
> ruri.c:1021:41: error: use of undeclared identifier '__SNR_futex_waitv'
>   seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(futex_waitv), 0);
>                                         ^
> ```
> 
> clang version:14.0.1

I tried to build it in termux and Kali Linux  amd64, but there's no errors. Can you tell me more information about your device?

**Comment by dabao1955** [comment]
> > ```
> > ruri.c:1021:41: error: use of undeclared identifier '__SNR_futex_waitv'
> >   seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(futex_waitv), 0);
> >                                         ^
> > ```
> > 
> > clang version:14.0.1
> 
> I tried to build it in termux and Kali Linux  amd64, but there's no errors. Can you tell me more information about your device?

Host: Ubuntu22.04.3 lts
kernel version:6.2.0

**Comment by Moe-hacker** [comment]
> ```
> ruri.c:1021:41: error: use of undeclared identifier '__SNR_futex_waitv'
>   seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(futex_waitv), 0);
>                                         ^
> ```
> 
> clang version:14.0.1

Well, I have just removed the error line to fix it.
I'm sure I followed docker's seccomp profile to write the code, but I cannot find `futex_waitv` syscall in it now. I think it's very strange because I can even pass the compiler on my devices.

**Comment by dabao1955** [comment]
who knows :(

________________________________
发件人: t[ ***@***.***>
发送时间: 2023年9月12日星期二 21:51
收件人: Moe-hacker/ruri ***@***.***>
抄送: 1955 dabao ***@***.***>; Author ***@***.***>
主题: Re: [Moe-hacker/ruri] Build err (Discussion #5)


ruri.c:1021:41: error: use of undeclared identifier '__SNR_futex_waitv'
  seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(futex_waitv), 0);
                                        ^


clang version:14.0.1

Well, I have just removed the error line to fix it.
I'm sure I followed docker's seccomp profile to write the code, but I cannot find futex_waitv syscall in it now. I think it's very strange because I can even pass the compiler on my devices.

―
